### PR TITLE
Add ADR for transformer architecture

### PR DESCRIPTION
**Pull request by danialmalik**
**Description:** 
Add ADR for decision about developing Transformers as "Processors"

**Reviewers:**
- [ ] tag reviewer 
- [ ] tag reviewer 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @danialmalik! I've created [OSPR-5076](https://openedx.atlassian.net/browse/OSPR-5076) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: docs/decisions/0004-transformers-architecture.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+If we decide to develop the event transformers as “processors”, it will result
+in less complexity in the code since the transformer can be easily appended in
+any backend’s (router or logger) processors’ pipeline.
+
+
+Decision
+--------
+
+Transformers will be developed as event transformers that can be added in
```
Comment (feanil) [inline docs/decisions/0004-transformers-architecture.md:25]:
```suggestion
Transformers will be developed as event processors that can be added in
```

**Review by feanil — CHANGES_REQUESTED** [review]
One change then I think this is good to go.

File: docs/decisions/0004-transformers-architecture.md (hunk 1 above)
Comment (danialmalik) [reply docs/decisions/0004-transformers-architecture.md]: My bad. Updated now :+1:

**Comment by natabene** [comment]
@danialmalik Thank you for your contribution!

**Comment by natabene** [comment]
@feanil Could you review this?

**Review by feanil — APPROVED** [review]

**Comment by edx-webhook** [comment]
@danialmalik 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/event-routing-backends&entry.867055334=https://github.com/edx/event-routing-backends/pull/10&entry.1484655318=https://github.com/danialmalik&entry.752974735=2020-10-22+12:33&entry.1917517419=2021-01-15+14:24&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**